.circleci/config.yml

```diff
@@ -477,6 +481,78 @@ jobs:
           fail_only: true
           failure_message: ":blobonfire: Nightly job failed to release"
 
+  pr-preview:
+    docker:
+      - image: circleci/python:3.8.1
+
+    resource_class: medium+
+
+    working_directory: ~/repo
+
+    steps:
+      - add_ssh_keys:
+          name: Add the read/write Github deploy key
+          fingerprints:
+            - "06:66:05:ea:ba:10:ec:b4:93:06:ab:62:7c:20:33:f0"
+
+      - checkout:
+          name: Checkout Streamlit code
+
+      - update-submodules
+
+      - pre-cache
+      - restore-from-cache
+      - pre-make
+      - make-init
+
+      - run:
+          <<: *make_scss_vars
```

Comment:
I see this comes from nightly but I don't think we need to run make scssvars. We do that as part of `make package`. Happy to clean that up as a separate item for here and nightly

Reply:
Though as I'm thinking through that, perhaps it makes sense to move the commonalities into a command. Then this job just focuses on uploading to S3 and setting up the GH branch, etc

Reply:
Yes these absolutely should be extracted to a common command, but I was also scared to upset the CircleCI gods. Let's do a follow-up.